src/contrib/mlir/lowerer.cpp

```diff
@@ -858,6 +876,51 @@ namespace
                     NGRAPH_CHECK(false, "Unsupported type for Negative");
                 }
             }
+            else if (isa<NGNotOp>(op))
+            {
+                iRes(ivs) = !val;
+            }
+            else if (isa<NGAbsOp>(op))
```

Comment:
Could we try to create a function overloaded for each op so that it's more like a visitor implementation? Something like:
```
void generate_op_edsc(NGNotOp *op, ...) {}
void generate_op_edsc(NGAbsOp *op, ...) {}
... 
```
Would that work?